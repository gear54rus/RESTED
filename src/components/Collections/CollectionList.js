import React from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';

import collectionPropType from 'propTypes/collection';
import { getCollections, collectionsAreFetching } from 'store/collections/selectors';
import * as Actions from 'store/collections/actions';

import Collection from './Collection';

class CollectionList extends React.Component {
  componentDidMount() {
    this.props.fetchCollections();
  }

  render() {
    const { collections, isFetching } = this.props;

    return (
      <span>
        {collections.map((collection, index) => (
          <Collection
            key={collection.id}
            collectionIndex={index}
            {...collection}
          />
        ))}
        {!isFetching && !collections.length && (
          <div>
            <h5>
              No collected requests.
              Add by pressing &quot;plus&quot; in the top right of the request panel.
            </h5>
          </div>
        )}
      </span>
    );
  }
}

CollectionList.propTypes = {
  collections: PropTypes.arrayOf(collectionPropType).isRequired,
  isFetching: PropTypes.bool.isRequired,
  fetchCollections: PropTypes.func.isRequired,
};

const mapStateToProps = state => ({
  collections: getCollections(state).toJS(),
  isFetching: collectionsAreFetching(state),
});

export default connect(mapStateToProps, Actions)(CollectionList);

