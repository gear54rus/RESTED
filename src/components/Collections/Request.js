import React from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import { DragSource, DropTarget } from 'react-dnd';
import { ListGroup } from 'react-bootstrap';
import flow from 'lodash.flow';

import IconButton from 'components/IconButton';
import * as CollectionActions from 'store/collections/actions';
import * as RequestActions from 'store/request/actions';
import * as ConfigActions from 'store/config/actions';
import { isDefaultCompact } from 'store/options/selectors';
import { getEditingRequest } from 'store/config/selectors';
import selectText from 'utils/selectText';

import { StyledRequest, RequestButtons, MainContentDiv } from './StyledComponents';
import * as Type from './dropTypes';

/**
 * Specifies the drag source contract.
 */
const requestSource = {
  canDrag(props) {
    // Disallow drag if editing request
    return !props.isEditing;
  },

  beginDrag({ request, index, collectionIndex }) {
    return { request, index, collectionIndex };
  },
};

const requestTarget = {
  hover(props, monitor) {
    const dragIndex = monitor.getItem().index;
    const dragCollectionIndex = monitor.getItem().collectionIndex;
    const dragUUID = monitor.getItem().request.id;
    const hoverIndex = props.index;
    const hoverCollectionIndex = props.collectionIndex;
    const hoverUUID = props.request.id;

    // Abort further processing if dragged item is hovering over itself
    if (dragUUID === hoverUUID) {
      return;
    }

    // Note: we're mutating the monitor item here.
    // Generally it's better to avoid mutations,
    // but it's good here for the sake of performance
    // to avoid expensive index searches.
    /* eslint-disable no-param-reassign */
    monitor.getItem().index = hoverIndex;
    monitor.getItem().collectionIndex = hoverCollectionIndex;
    /* eslint-enable no-param-reassign */

    // Update redux orders
    props.reorderRequest({
      // Source
      collectionIndex: dragCollectionIndex,
      requestIndex: dragIndex,
    }, {
      // Target
      collectionIndex: hoverCollectionIndex,
      requestIndex: hoverIndex,
    });
  },
};

class Request extends React.Component {
  static propTypes = {
    connectDragSource: PropTypes.func.isRequired,
    connectDropTarget: PropTypes.func.isRequired,
    isDragging: PropTypes.bool.isRequired,
    index: PropTypes.number.isRequired,
    collectionIndex: PropTypes.number.isRequired,
    renameRequest: PropTypes.func.isRequired,
    selectRequest: PropTypes.func.isRequired,
    sendRequest: PropTypes.func.isRequired,
    deleteRequest: PropTypes.func.isRequired,
    toggleEditMode: PropTypes.func.isRequired,
    isEditing: PropTypes.bool.isRequired,
    defaultCompact: PropTypes.bool.isRequired,
    request: PropTypes.shape({
      id: PropTypes.string.isRequired,
      name: PropTypes.string,
      method: PropTypes.string.isRequired,
      url: PropTypes.string.isRequired,
    }).isRequired,
  };

  state = {
    compact: this.props.defaultCompact,
  };

  toggleCompact = () => {
    this.setState({ compact: !this.state.compact });
  }

  toggleEdit = () => {
    this.props.toggleEditMode(this.props.request);
  }

  renameRequest = e => {
    const { collectionIndex, index, renameRequest } = this.props;
    e.preventDefault();

    this.toggleEdit();
    renameRequest(collectionIndex, index, this.nameRef.value);
  }

  renderRequest() {
    const { compact } = this.state;
    const { request, index, collectionIndex, isEditing } = this.props;
    const { method, name, url } = request;

    return (
      <div>
        {!compact && <h4>{method}</h4>}
        {isEditing ? (
          <form onSubmit={this.renameRequest}>
            <label
              className="sr-only"
              htmlFor={`${collectionIndex}.${index}.RequestName`}
            >
              Request name
            </label>
            <input
              autoFocus
              onFocus={selectText}
              id={`${collectionIndex}.${index}.RequestName`}
              defaultValue={name}
              ref={ref => { this.nameRef = ref; }}
            />
            <IconButton
              tooltip="Save"
              icon="check"
            />
          </form>
        ) : (
          name || url
        )}
      </div>
    );
  }

  render() {
    const { compact } = this.state;
    const {
      connectDragSource,
      connectDropTarget,
      isDragging,
      request,
      collectionIndex,
      selectRequest,
      sendRequest,
      deleteRequest,
      isEditing,
    } = this.props;

    return connectDragSource(connectDropTarget(
      <div>{/* Need a wrapper div for React DnD support */}
        <StyledRequest isDragging={isDragging}>
          <ListGroup componentClass="div">
            <div className="list-group-item">
              <RequestButtons compact={compact}>
                {!compact && (
                  <IconButton
                    tooltip="Toggle edit"
                    icon="pencil-alt"
                    onClick={this.toggleEdit}
                  />
                )}
                {compact && (
                  <IconButton
                    tooltip="Expand"
                    icon="plus"
                    onClick={this.toggleCompact}
                  />
                )}

                {!compact && !isEditing && (
                  <IconButton
                    tooltip="Minimize"
                    icon="minus"
                    onClick={this.toggleCompact}
                  />
                )}
                {!compact && isEditing && (
                  <IconButton
                    tooltip="Delete"
                    icon="trash"
                    onClick={() => deleteRequest(request.id, collectionIndex)}
                  />
                )}
              </RequestButtons>

              {isEditing ? (
                <MainContentDiv compact={compact}>
                  {this.renderRequest()}
                </MainContentDiv>
              ) : (
                <MainContentDiv
                  compact={compact}
                  onClick={() => selectRequest(request)}
                  onDoubleClick={() => sendRequest(request)}
                >
                  {this.renderRequest()}
                </MainContentDiv>
              )}
            </div>
          </ListGroup>
        </StyledRequest>
      </div>,
    ));
  }
}

const mapStateToProps = (state, props) => ({
  isEditing: getEditingRequest(state)
    ? getEditingRequest(state).id === props.request.id
    : false,
  defaultCompact: isDefaultCompact(state),
});

export default flow(
  DropTarget(Type.Request, requestTarget, connector => ({
    connectDropTarget: connector.dropTarget(),
  })),
  DragSource(Type.Request, requestSource, (connector, monitor) => ({
    connectDragSource: connector.dragSource(),
    isDragging: monitor.isDragging(),
  })),
  connect(mapStateToProps, {
    ...CollectionActions,
    ...RequestActions,
    ...ConfigActions,
  }),
)(Request);

