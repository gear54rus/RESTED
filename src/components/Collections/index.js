import React from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import { Panel } from 'react-bootstrap';

import IconButton from 'components/IconButton';
import * as Actions from 'store/collections/actions';

import CollectionList from './CollectionList';

function Titlebar({ addCollection }) {
  return (
    <span className="clearfix">
      <IconButton
        onClick={() => addCollection()}
        tooltip="Add new collection"
        icon="plus"
        className="pull-right"
      />
    </span>
  );
}

Titlebar.propTypes = {
  addCollection: PropTypes.func.isRequired,
};

function Collections({ addCollection }) {
  return (
    <Panel>
      <Panel.Heading>
        <Panel.Title
          componentClass={Titlebar}
          addCollection={addCollection}
        />
      </Panel.Heading>
      <Panel.Body>
        <CollectionList />
      </Panel.Body>
    </Panel>
  );
}

Collections.propTypes = {
  addCollection: PropTypes.func.isRequired,
};

export default connect(null, Actions)(Collections);

