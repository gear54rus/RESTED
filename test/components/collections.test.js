import React from 'react';
import { Provider } from 'react-redux';
import renderer from 'react-test-renderer';
import { mount } from 'enzyme';
import { DragDropContext } from 'react-dnd';
import TestBackend from 'react-dnd-test-backend';

/* eslint-disable import/no-unresolved */
import Collections from 'components/Collections';
import Collection from 'components/Collections/Collection';

import makeStore from '../makeStore';
import mockCollections from '../mock/collections.mock';

jest.mock('react-dom');

/**
 * Wraps a component into a DragDropContext that uses the TestBackend.
 */
function wrapInTestContext(DecoratedComponent, store = makeStore()) {
  return DragDropContext(TestBackend)(
    /* eslint-disable react/prefer-stateless-function */
    class TestContextContainer extends React.Component {
      render() {
        return (
          <Provider store={store}>
            <DecoratedComponent {...this.props} />
          </Provider>
        );
      }
    },
    /* eslint-enable react/prefer-stateless-function */
  );
}

describe('Collections component', () => {
  const store = makeStore({
    collections: mockCollections,
  });

  it('should match the previous snapshot', () => {
    const BoxedComponent = wrapInTestContext(Collections, store);
    const tree = renderer.create(
      <BoxedComponent />,
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('should render a "no requests" message when given no collections', () => {
    const BoxedComponent = wrapInTestContext(Collections);
    const tree = mount(
      <BoxedComponent />,
    );

    expect(tree.find('h5').prop('children')).toEqual('No collected requests. Add by pressing "plus" in the top right of the request panel.');
  });

  it('should render two collections when given a dataset with two collections', () => {
    const BoxedComponent = wrapInTestContext(Collections, store);
    const tree = mount(
      <BoxedComponent />,
    );

    expect(tree.find('Collection').length).toEqual(2);
  });
});

describe('Collection component', () => {
  it('should match the previous snapshot', () => {
    const BoxedComponent = wrapInTestContext(Collection);
    const requests = mockCollections
      .getIn(['collections', 0, 'requests'])
      .toJS();

    const tree = renderer.create(
      <BoxedComponent
        collectionIndex={0}
        requests={requests}
      />,
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });

  xit('should change the styles of dragged elements');
  /*
    const BoxedComponent = wrapInTestContext(Collection);
    const requests = mockCollections
      .getIn(['collections', 1, 'requests'])
      .toJS();

    const tree = mount(
      <BoxedComponent
        collectionIndex={1}
        requests={requests}
      />
    );

    // Obtain a reference to the backend
    const backend = tree.instance().getManager().getBackend();

    // Test that the background color is white
    expect(tree.find('Panel').props.style.opacity).toEqual(1);

    // Find the drag source ID and use it to simulate the dragging operation
    const box = TestUtils.findRenderedComponentWithType(root, Box);
    backend.simulateBeginDrag([box.getHandlerId()]);

    // Verify that the element changed its color
    const div = TestUtils.findRenderedDOMComponentWithTag(root, 'div');
    expect(div.props.style.opacity).toEqual(0.4);

    // See other backend.simulate* methods for more!
  });
  */

  xit('should have draggable requests');
});

