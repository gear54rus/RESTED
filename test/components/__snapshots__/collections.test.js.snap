// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Collection component should match the previous snapshot 1`] = `
<div>
  <div
    className="sc-ifAKCX lgbDDf"
  >
    <span
      className="sc-EHOje vEhJb"
    >
      <h3
        onClick={[Function]}
      />
      <button
        className="pull-right sc-bdVaJa cORElC"
        onClick={[Function]}
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
      >
        <div
          className="sc-bwzfXH dwTRXS"
        >
          <div
            className="sc-htpNat kDKKFi"
          />
          <span
            className="sc-bxivhb kOilMN"
          >
            Delete
          </span>
        </div>
        <i
          className="fas fa-trash"
          role="presentation"
        />
      </button>
      <button
        className="pull-right sc-bdVaJa cORElC"
        onClick={[Function]}
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
      >
        <div
          className="sc-bwzfXH dwTRXS"
        >
          <div
            className="sc-htpNat kDKKFi"
          />
          <span
            className="sc-bxivhb kOilMN"
          >
            Change name
          </span>
        </div>
        <i
          className="fas fa-pencil-alt"
          role="presentation"
        />
      </button>
      <hr />
    </span>
    <div
      aria-expanded={null}
      className="collapse in"
    >
      <div>
        <div
          className="sc-bZQynM jwpJAK"
        >
          <div
            className="list-group"
          >
            <div
              className="list-group-item"
            >
              <div
                className="sc-gzVnrw bBTGUh"
              >
                <button
                  className="sc-bdVaJa cORElC"
                  onClick={[Function]}
                  onMouseEnter={[Function]}
                  onMouseLeave={[Function]}
                >
                  <div
                    className="sc-bwzfXH dwTRXS"
                  >
                    <div
                      className="sc-htpNat kDKKFi"
                    />
                    <span
                      className="sc-bxivhb kOilMN"
                    >
                      Toggle edit
                    </span>
                  </div>
                  <i
                    className="fas fa-pencil-alt"
                    role="presentation"
                  />
                </button>
                <button
                  className="sc-bdVaJa cORElC"
                  onClick={[Function]}
                  onMouseEnter={[Function]}
                  onMouseLeave={[Function]}
                >
                  <div
                    className="sc-bwzfXH dwTRXS"
                  >
                    <div
                      className="sc-htpNat kDKKFi"
                    />
                    <span
                      className="sc-bxivhb kOilMN"
                    >
                      Minimize
                    </span>
                  </div>
                  <i
                    className="fas fa-minus"
                    role="presentation"
                  />
                </button>
              </div>
              <div
                className="sc-htoDjs ieWywL"
                onClick={[Function]}
                onDoubleClick={[Function]}
              >
                <div>
                  <h4>
                    POST
                  </h4>
                  www.foo.no
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Collections component should match the previous snapshot 1`] = `
<div
  className="panel panel-default"
>
  <div
    className="panel-heading"
  >
    <span
      className="clearfix"
    >
      <button
        className="pull-right sc-bdVaJa cORElC"
        onClick={[Function]}
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
      >
        <div
          className="sc-bwzfXH dwTRXS"
        >
          <div
            className="sc-htpNat kDKKFi"
          />
          <span
            className="sc-bxivhb kOilMN"
          >
            Add new collection
          </span>
        </div>
        <i
          className="fas fa-plus"
          role="presentation"
        />
      </button>
    </span>
  </div>
  <div
    className="panel-body"
  >
    <span>
      <div>
        <div
          className="sc-ifAKCX lgbDDf"
        >
          <span
            className="sc-EHOje bhfhJk"
          >
            <h3
              onClick={[Function]}
            >
              Collection
            </h3>
            <button
              className="pull-right sc-bdVaJa cORElC"
              onClick={[Function]}
              onMouseEnter={[Function]}
              onMouseLeave={[Function]}
            >
              <div
                className="sc-bwzfXH dwTRXS"
              >
                <div
                  className="sc-htpNat kDKKFi"
                />
                <span
                  className="sc-bxivhb kOilMN"
                >
                  Delete
                </span>
              </div>
              <i
                className="fas fa-trash"
                role="presentation"
              />
            </button>
            <button
              className="pull-right sc-bdVaJa cORElC"
              onClick={[Function]}
              onMouseEnter={[Function]}
              onMouseLeave={[Function]}
            >
              <div
                className="sc-bwzfXH dwTRXS"
              >
                <div
                  className="sc-htpNat kDKKFi"
                />
                <span
                  className="sc-bxivhb kOilMN"
                >
                  Change name
                </span>
              </div>
              <i
                className="fas fa-pencil-alt"
                role="presentation"
              />
            </button>
            <hr />
          </span>
          <div
            aria-expanded={null}
            className="collapse"
          >
            <div>
              <div
                className="sc-bZQynM jwpJAK"
              >
                <div
                  className="list-group"
                >
                  <div
                    className="list-group-item"
                  >
                    <div
                      className="sc-gzVnrw bBTGUh"
                    >
                      <button
                        className="sc-bdVaJa cORElC"
                        onClick={[Function]}
                        onMouseEnter={[Function]}
                        onMouseLeave={[Function]}
                      >
                        <div
                          className="sc-bwzfXH dwTRXS"
                        >
                          <div
                            className="sc-htpNat kDKKFi"
                          />
                          <span
                            className="sc-bxivhb kOilMN"
                          >
                            Toggle edit
                          </span>
                        </div>
                        <i
                          className="fas fa-pencil-alt"
                          role="presentation"
                        />
                      </button>
                      <button
                        className="sc-bdVaJa cORElC"
                        onClick={[Function]}
                        onMouseEnter={[Function]}
                        onMouseLeave={[Function]}
                      >
                        <div
                          className="sc-bwzfXH dwTRXS"
                        >
                          <div
                            className="sc-htpNat kDKKFi"
                          />
                          <span
                            className="sc-bxivhb kOilMN"
                          >
                            Minimize
                          </span>
                        </div>
                        <i
                          className="fas fa-minus"
                          role="presentation"
                        />
                      </button>
                    </div>
                    <div
                      className="sc-htoDjs ieWywL"
                      onClick={[Function]}
                      onDoubleClick={[Function]}
                    >
                      <div>
                        <h4>
                          POST
                        </h4>
                        www.foo.no
                      </div>
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div>
        <div
          className="sc-ifAKCX lgbDDf"
        >
          <span
            className="sc-EHOje bhfhJk"
          >
            <h3
              onClick={[Function]}
            >
              Collection 2
            </h3>
            <button
              className="pull-right sc-bdVaJa cORElC"
              onClick={[Function]}
              onMouseEnter={[Function]}
              onMouseLeave={[Function]}
            >
              <div
                className="sc-bwzfXH dwTRXS"
              >
                <div
                  className="sc-htpNat kDKKFi"
                />
                <span
                  className="sc-bxivhb kOilMN"
                >
                  Delete
                </span>
              </div>
              <i
                className="fas fa-trash"
                role="presentation"
              />
            </button>
            <button
              className="pull-right sc-bdVaJa cORElC"
              onClick={[Function]}
              onMouseEnter={[Function]}
              onMouseLeave={[Function]}
            >
              <div
                className="sc-bwzfXH dwTRXS"
              >
                <div
                  className="sc-htpNat kDKKFi"
                />
                <span
                  className="sc-bxivhb kOilMN"
                >
                  Change name
                </span>
              </div>
              <i
                className="fas fa-pencil-alt"
                role="presentation"
              />
            </button>
            <hr />
          </span>
          <div
            aria-expanded={null}
            className="collapse"
          >
            <div>
              <div
                className="sc-bZQynM jwpJAK"
              >
                <div
                  className="list-group"
                >
                  <div
                    className="list-group-item"
                  >
                    <div
                      className="sc-gzVnrw bBTGUh"
                    >
                      <button
                        className="sc-bdVaJa cORElC"
                        onClick={[Function]}
                        onMouseEnter={[Function]}
                        onMouseLeave={[Function]}
                      >
                        <div
                          className="sc-bwzfXH dwTRXS"
                        >
                          <div
                            className="sc-htpNat kDKKFi"
                          />
                          <span
                            className="sc-bxivhb kOilMN"
                          >
                            Toggle edit
                          </span>
                        </div>
                        <i
                          className="fas fa-pencil-alt"
                          role="presentation"
                        />
                      </button>
                      <button
                        className="sc-bdVaJa cORElC"
                        onClick={[Function]}
                        onMouseEnter={[Function]}
                        onMouseLeave={[Function]}
                      >
                        <div
                          className="sc-bwzfXH dwTRXS"
                        >
                          <div
                            className="sc-htpNat kDKKFi"
                          />
                          <span
                            className="sc-bxivhb kOilMN"
                          >
                            Minimize
                          </span>
                        </div>
                        <i
                          className="fas fa-minus"
                          role="presentation"
                        />
                      </button>
                    </div>
                    <div
                      className="sc-htoDjs ieWywL"
                      onClick={[Function]}
                      onDoubleClick={[Function]}
                    >
                      <div>
                        <h4>
                          POST
                        </h4>
                        www.foo.no
                      </div>
                    </div>
                  </div>
                </div>
              </div>
            </div>
            <div>
              <div
                className="sc-bZQynM jwpJAK"
              >
                <div
                  className="list-group"
                >
                  <div
                    className="list-group-item"
                  >
                    <div
                      className="sc-gzVnrw bBTGUh"
                    >
                      <button
                        className="sc-bdVaJa cORElC"
                        onClick={[Function]}
                        onMouseEnter={[Function]}
                        onMouseLeave={[Function]}
                      >
                        <div
                          className="sc-bwzfXH dwTRXS"
                        >
                          <div
                            className="sc-htpNat kDKKFi"
                          />
                          <span
                            className="sc-bxivhb kOilMN"
                          >
                            Toggle edit
                          </span>
                        </div>
                        <i
                          className="fas fa-pencil-alt"
                          role="presentation"
                        />
                      </button>
                      <button
                        className="sc-bdVaJa cORElC"
                        onClick={[Function]}
                        onMouseEnter={[Function]}
                        onMouseLeave={[Function]}
                      >
                        <div
                          className="sc-bwzfXH dwTRXS"
                        >
                          <div
                            className="sc-htpNat kDKKFi"
                          />
                          <span
                            className="sc-bxivhb kOilMN"
                          >
                            Minimize
                          </span>
                        </div>
                        <i
                          className="fas fa-minus"
                          role="presentation"
                        />
                      </button>
                    </div>
                    <div
                      className="sc-htoDjs ieWywL"
                      onClick={[Function]}
                      onDoubleClick={[Function]}
                    >
                      <div>
                        <h4>
                          POST
                        </h4>
                        www.bar.no
                      </div>
                    </div>
                  </div>
                </div>
              </div>
            </div>
            <div>
              <div
                className="sc-bZQynM jwpJAK"
              >
                <div
                  className="list-group"
                >
                  <div
                    className="list-group-item"
                  >
                    <div
                      className="sc-gzVnrw bBTGUh"
                    >
                      <button
                        className="sc-bdVaJa cORElC"
                        onClick={[Function]}
                        onMouseEnter={[Function]}
                        onMouseLeave={[Function]}
                      >
                        <div
                          className="sc-bwzfXH dwTRXS"
                        >
                          <div
                            className="sc-htpNat kDKKFi"
                          />
                          <span
                            className="sc-bxivhb kOilMN"
                          >
                            Toggle edit
                          </span>
                        </div>
                        <i
                          className="fas fa-pencil-alt"
                          role="presentation"
                        />
                      </button>
                      <button
                        className="sc-bdVaJa cORElC"
                        onClick={[Function]}
                        onMouseEnter={[Function]}
                        onMouseLeave={[Function]}
                      >
                        <div
                          className="sc-bwzfXH dwTRXS"
                        >
                          <div
                            className="sc-htpNat kDKKFi"
                          />
                          <span
                            className="sc-bxivhb kOilMN"
                          >
                            Minimize
                          </span>
                        </div>
                        <i
                          className="fas fa-minus"
                          role="presentation"
                        />
                      </button>
                    </div>
                    <div
                      className="sc-htoDjs ieWywL"
                      onClick={[Function]}
                      onDoubleClick={[Function]}
                    >
                      <div>
                        <h4>
                          POST
                        </h4>
                        www.baz.no
                      </div>
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </span>
  </div>
</div>
`;
